src/modules/Channel/components/MessageList/index.tsx

```diff
@@ -152,6 +154,13 @@ const MessageList: React.FC<MessageListProps> = ({
 
   const { scrollToBottomHandler, scrollBottom } = useSetScrollToBottom({ loading });
 
+  const renderedFrozenNotification = useMemo(() => {
+    if (renderFrozenNotification) {
+      return renderFrozenNotification();
+    }
+    return null;
+  }, [renderFrozenNotification]);
+
```

Comment:
It seems okay to remove the useMemo.

Reply:
Is it because renderedFrozenNotification does not change often?

Reply:
Here are several reasons:

- In typical cases (when `renderFrozenNotification` is not passed), useMemo is unnecessary.
- If `renderFrozenNotification` is passed inline, useMemo has no effect. (always received new function)
- If `renderFrozenNotification` is passed using useCallback, there might be some optimization for component rendering. However, if optimization for component rendering is necessary, it's recommended to use `React.memo` on the side implementing the custom interface rather than using useMemo from our side.

Reply:
Nice reasoning!